### PR TITLE
Ensure logs are flushed

### PR DESCRIPTION
**Pull request by crosbymichael**
This ensures that anything written to the logs are synced as they
happen.

This also changes the error message of the libcontainer error.  The
original idea was to have this extra information in the message but it
makes it hard to parse and if the caller needed this information they
can just get it from the error type.

Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -88,7 +89,7 @@ func main() {
 			logrus.SetLevel(logrus.DebugLevel)
 		}
 		if path := context.GlobalString("log"); path != "" {
-			f, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
+			f, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY|os.O_APPEND|syscall.O_DSYNC, 0666)
```
Comment (hqhq) [inline main.go]: Shall we just use `os.O_SYNC`? Looks more consistent.

Comment (crosbymichael) [inline main.go]: yes, thanks

**Comment by hqhq** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM
